src/libraries/navdrawer/navdrawer.js

```diff
@@ -3,358 +3,373 @@
+        if (this.isPeeking) {
+            this.isPeeking = false;
+            dom.removeEventListener(this.edgeContainer, 'touchmove', this.onEdgeTouchMove, {});
+            this.onMenuTouchEnd(e);
         }
-    }
-
-    function disableEvent(e) {
-        e.preventDefault();
-        e.stopPropagation();
-    }
+    };
 
-    function onBackgroundTouchStart(e) {
+    onBackgroundTouchStart = (e) => {
         const touches = getTouches(e);
         const touch = touches[0] || {};
-        backgroundTouchStartX = touch.clientX;
-        backgroundTouchStartTime = new Date().getTime();
-    }
+        this.backgroundTouchStartX = touch.clientX;
+        this.backgroundTouchStartTime = new Date().getTime();
+    };
+
+    onBackgroundTouchMove = (e) => {
+        const options = this.options;
 
-    function onBackgroundTouchMove(e) {
         const touches = getTouches(e);
         const touch = touches[0] || {};
         const endX = touch.clientX || 0;
 
-        if (endX <= options.width && self.isVisible) {
-            countStart++;
-            const deltaX = endX - (backgroundTouchStartX || 0);
+        if (endX <= options.width && this.isVisible) {
+            this.countStart++;
+            const deltaX = endX - (this.backgroundTouchStartX || 0);
 
-            if (countStart == 1) {
-                startPoint = deltaX;
+            if (this.countStart == 1) {
+                this.startPoint = deltaX;
             }
-            if (deltaX < 0 && dragMode !== 2) {
-                dragMode = 1;
-                newPos = deltaX - startPoint + options.width;
-                self.changeMenuPos();
-                const time = new Date().getTime() - (backgroundTouchStartTime || 0);
-                velocity = Math.abs(deltaX) / time;
+            if (deltaX < 0 && this.dragMode !== 2) {
+                this.dragMode = 1;
+                this.newPos = deltaX - this.startPoint + options.width;
+                this.changeMenuPos();
+                const time = new Date().getTime() - (this.backgroundTouchStartTime || 0);
+                this.velocity = Math.abs(deltaX) / time;
             }
         }
 
         e.preventDefault();
         e.stopPropagation();
-    }
+    };
 
-    function onBackgroundTouchEnd(e) {
+    onBackgroundTouchEnd = (e) => {
         const touches = getTouches(e);
         const touch = touches[0] || {};
         const endX = touch.clientX || 0;
-        const deltaX = endX - (backgroundTouchStartX || 0);
-        self.checkMenuState(deltaX);
-        countStart = 0;
-    }
+        const deltaX = endX - (this.backgroundTouchStartX || 0);
+        this.checkMenuState(deltaX);
+        this.countStart = 0;
+    };
 
-    function onMaskTransitionEnd() {
-        const classList = mask.classList;
+    onMaskTransitionEnd = () => {
+        const classList = this.mask.classList;
 
         if (!classList.contains('backdrop')) {
             classList.add('hide');
         }
-    }
```

Comment:
It looks fine, but I can't get it to be called.
We can probably remove this. Maybe not now.

Reply:
this was called when opacity changed
```
.tmla-mask.backdrop {
    opacity: 1;
}
```